docs/developer_guidelines.md

```diff
@@ -124,68 +126,81 @@ Resources:
 - [One Idea is One Commit] - Interesting committing strategy, comparable to
   atomic commits.
 
-Use the `gitlint` tool to check your commit messages. See
-[tools/gitlint/README.md] for more information.
+### Message style and convention
 
-### Message style
+We follow [Conventional Commits] with a minor deviation:
 
-There are two possible styles for the commit messages: short and full.
+- since we do not usually create libraries, everything related to the
+  `BREAKING-CHANGE` and `!` rule is ignored. Specifically ignore rules 11., 12.,
+  13., 15., and 16. of the Conventional Commits [specification].
 
-#### Short
+[Conventional Commits]: https://www.conventionalcommits.org/
+[specification]: https://www.conventionalcommits.org/en/v1.0.0/#specification
 
-Short commit messages should be around 50 characters long, the maximum is 72.
+We use [`commited`] tool (run by the `pre-commit`) to validate commit messages.
+For relevant configuration see [`commited.toml`] file in our template
+repository.
 
-Use this when the change that you made is either:
+[`commited`]: https://github.com/crate-ci/committed
+[`commited.toml`]:
+  https://github.com/IRNAS/irnas-zephyr-template/blob/main/commited.toml
 
-- self-explanatory, for example: `Fix typo in README.md` or
-- there is a GitHub issue that provides sufficient context for the change made.
+#### Structure and rules
 
-Mention relevant GitHub issues at the end of the message, for example:
+The commit message should be structured as follows:
 
-- `Fix out of bounds bug, closes #123`
-- `Add shell module, relevant #456`
+```code
+<type>[optional scope]: <subject>
 
-#### Full
+[optional body]
 
-Whenever the change you made needs more context you should use the full message
-format.
+[optional footer(s)]
+```
 
-The full message format is best summarized with the below rules:
+General rules that must be followed:
 
-- Separate the subject from the body with a blank line.
-- Limit the subject line to 50 characters.
-- Capitalize the subject line.
+- Every commit message must have a type and a subject. Scope, body and footer
+  sections are optional.
+- Separate the subject from the body with a blank line. The subject line should
+  be around 50 characters, the maximum is 72.
 - Do not end the subject line with a period.
 - Use the imperative mood in the subject line.
 - Wrap the body at 72 characters.
```

Comment:
```suggestion
- Wrap the body at 80 characters.
```

The Gitlint configuration we had used previously was 80 chars for the body. I see this was wrong from the start. I have no opinion on enforcing 72 now.

Reply:
You are correct, I assumed that the default is set to the 72, because I actually never broke it (default behavior of my editor is to wrap at 72 in commit messages), but on the further inspection of the gitlint configuration rules the body really is set to 80: https://jorisroovers.com/gitlint/latest/rules/builtin_rules/#b1-body-max-line-length

I meant to use 72 initially, so this will now be correctly enforced by the `commited` now.